pkg/tools/etcd_tools.go

```diff
@@ -359,10 +409,11 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 			return nil
 		}
 
-		_, err = h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
+		response, err := h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
```

Comment:
It's more for things like "there should be a thing here with these values, it's ok if a conflict happens and I overwrite it because of the semantics of the request".  Resize to a specific number (absolute) is probably more efficiently implemented this way, where you say "size = X" and last write wins.

----- Original Message -----
> > @@ -359,10 +409,11 @@ func (h *EtcdHelper) AtomicUpdate(key string,
> > ptrToType runtime.Object, ignoreNo
> >  			return nil
> >  		}
> >  
> > -		_, err = h.Client.CompareAndSwap(key, string(data), 0, origBody, index)
> > +		response, err := h.Client.CompareAndSwap(key, string(data), 0, origBody,
> > index)
> 
> Blind PUT with no precondition? I don't know how important that is. Esp.
> since we store Spec and Status together, I don't even think we should allow
> that at the storage layer.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4419/files#r24686782